charter.md

```diff
@@ -53,15 +53,207 @@ The following responsibilities belong to the Steering Committe, but may be deleg
+
+## Membership
+
+### Composition
+
+The Steering Committee is composed of seven (7) members.
+
+### Resignation
+
+If a committee member chooses not to continue in their role, for whatever
+self-elected reason, they must notify the committee in writing. As a courtesy,
+such notifications should be given at least 30 calendar days in advance of
+their departure.
+
+The committee can choose to accept a member's resignation and carry out
+offboarding procedures at any time following the written notice.
+
+### Removal
+
+<!--
+
+STEERING CONSIDERATIONS
+
+Removal
+
+- [ ] Special election for specific removal scenarios
+  - What is the perception of having a committee remove a representative that was elected by the community?
+- [ ] Codify the voting process and channels for removal scenarios
+- [ ] Consider CoCC guidance on removals as prior art: https://github.com/kubernetes/community/blob/master/committee-code-of-conduct/charter.md#removal
+- [ ] Consider prior art in changes to maintainership e.g., SPIFFE/SPIRE (https://github.com/spiffe/spire/blob/main/MAINTAINERS.md#changes-in-maintainership)
+
+COMMUNITY CONSIDERATIONS
+
+No Confidence
+
+- [ ] Provide guidelines on "no confidence"
+  - Process transparency could be considered inversely proportional to the importance of these guidelines
+
+Removal
+
+- [ ] Scope of removal of members (one, many)
+- [ ] Precedence for removal scenarios
+- [ ] What details of removal scenarios should be public/private?
+  - Call for removal?
+  - Reason for removal?
+  - Vote on removal?
+  - What are the public artifacts of the removal process?
+- [ ] What details of election results should be public/private?
+- [ ] Consequences of removal scenarios
+  - Are individuals ineligible for re-election?
+    - Should community members have an opinion in this decision?
+- [ ] When is removal appropriate?
+- [ ] How should CoC concerns be considered in the removal process? Should they be?
+- [ ] Guard against "popularity contests" or "burning at the person" in committee voting scenarios
+- [ ] Are there antitrust implications in the removal process?
+- [ ] Special elections may not have representative turnout
+  - Define guidance on acts and thresholds to consider in a vote for removal
+- [ ] Is removal an extraordinary power to grant?
+  - "Policies like this have to work for the good of the community both when used well and when used poorly."
+- [ ] Define removal process guidelines
+  - "brief rationale for removal" + short time + "brief rebuttal"
+  - Offer resignation as an option
+
+-->
+
+### No confidence
+
+In the event a Steering Committee member receives a vote of no confidence from
+two-thirds of the remaining Steering Committee members, that person must either
+resign or call a special election in which the project community will vote on
+whether that person remains or is removed from the Steering Committee. A
+majority of eligible voters participating in the election must vote in favor of
```

Comment:
This special election (I think "recall" is a better word) sounds very expensive, personally I'd just require unanimity from the board (that's # permanent seats - 1 yeses), notice to the community with some vague rationale, and time between notice and the meeting where the vote is held, at least two weeks. More discussion about this is in another comment thread.

Reply:
Also due to our voting system, 50% of voters voting to retain a board member is quite likely a much higher fraction than what got them elected in the first place. If you keep this recall election, I think I would flip it around and make it a supermajority: 60% must vote in favor of ejection.